kratos/includes/registry_item.h

```diff
@@ -228,11 +292,39 @@ class KRATOS_API(KRATOS_CORE) RegistryItem
 
     RegistryItem& GetItem(std::string const& rItemName);
 
-    template<typename TDataType> TDataType const& GetValue() const
+    template <typename TDataType>
+    void EvaluateValue()
     {
         KRATOS_TRY
 
-        return *(std::any_cast<std::shared_ptr<TDataType>>(mpValue));
+        using TFunctionReturnType = std::shared_ptr<TDataType>;
+        using TFunctionType = std::function<TFunctionReturnType()>;
+
+        // Assign callable value
+        TFunctionType func = std::any_cast<TFunctionType>(mpValue);
+        mpCallable = func();
+
+        // Set value name
+        std::stringstream buffer;
+        buffer << *(std::any_cast<TFunctionReturnType>(mpCallable));
```

Comment:
the naming is **very** confusing. To my understanding, `mpCallable` is not a callable, but a pointer to the stored type. On the other hand, `mpValue` **is** actually a callable `std::function` instance and not a pointer, let alone a pointer to the stored type.

Reply:
At the very beginning I though the same... then we got used and we leave it like this (`mpCallable` is the thing to be always called once we've stored the prototype and `mpValue` is the  very first function that creates the prototype when first called).

I agree that it would be good to change it before merging it, but I'd leave this as the very last aesthetic change. Makes sense for you?